src/classes/groundplaces.ts

```diff
@@ -539,48 +660,132 @@ export class GroundPlacesController {
+      type: ActionType.DELETE_STOP_CLUSTER,
+    });
   }
 
   /**
-   * @description Getter to retrieve the Ground places.
+   * @description Getter to retrieve the GroundPlaces object.
    * @returns {GroundPlace[]}
    */
   public getGroundPlaces(): GroundPlace[] {
     return this.storageService.getGroundPlaces();
   }
 
   /**
-   * @description Download the GroundPlacesDiff file in JSON and store it on Desktop.
-   * @returns {void}
+   * @description Getter to retrieve the GroundPlacesDiff object.
+   * @returns {GroundPlaceDiff[]}
    */
-  public downloadGroundPlacesDiffToDesktop(): void {}
+  public getGroundPlacesDiff(): GroundPlaceDiff[] {
+    return this.storageService.getGroundPlacesDiff();
+  }
 
   /**
-   * @description Download the GroundPlaces file in JSON and store it on Desktop.
+   * @description Apply the diff file to the GroundPlaces object.
+   * @param {GroundPlaceDiff[]} groundPlacesDiff - File that store changes to apply on the GroundPlacesFile.
    * @returns {void}
    */
-  public downloadGroundPlacesFileToDesktop(): void {}
+  public applyGroundPlacesDiff(groundPlacesDiff: GroundPlaceDiff[]): void {
+    const groundPlaces: GroundPlace[] = this.getGroundPlaces();
+    const cloneGroundPlaces: GroundPlace[] = cloneDeep(this.getGroundPlaces());
+
+    if (!cloneGroundPlaces.length) {
+      throw new Error(
+        `You can't apply your GroundPlacesDiffFile because there is no GroundPlaces available on this instance. You should call the "init" method with your GroundPlacesFile before using this method.`,
+      );
+    }
+
+    try {
+      groundPlacesDiff.map((groundPlaceDiff: GroundPlaceDiff) => {
+        const [groundPlaceGpuid, { type, into: intoGroundPlaceGpuid, from: fromGroundPlaceGpuid, params }]: [
+          Gpuid,
+          GroundPlaceDiffOptions,
+        ] = Object.entries(groundPlaceDiff)[0];
+
+        switch (type) {
+          case ActionType.CREATE_STOP_GROUP:
+            return this.createStopGroup(params.segmentProviderStopId, groundPlaceGpuid, {
+              countryCode: params.countryCode,
+              latitude: params.latitude,
+              longitude: params.longitude,
+              name: params.name,
+            });
+
+          case ActionType.CREATE_STOP_CLUSTER:
+            return this.createStopCluster(groundPlaceGpuid, {
+              countryCode: params.countryCode,
+              latitude: params.latitude,
+              longitude: params.longitude,
+              name: params.name,
+            });
+
+          case ActionType.UPDATE_STOP_GROUP:
+            return this.updateStopGroup(groundPlaceGpuid, {
+              latitude: params.latitude,
+              longitude: params.longitude,
+              name: params.name,
+            });
+
+          case ActionType.UPDATE_STOP_CLUSTER:
+            return this.updateStopCluster(groundPlaceGpuid, {
+              latitude: params.latitude,
+              longitude: params.longitude,
```

Comment:
Carefull we don't want to send "undefined" i think

Reply:
Inside the method, I'll check for `undefined` values and only use the properties that are correctly setted, so this will works

Reply:
Fixed by `sanitize undefined values` before passing them to the `updateStopGroup` and `updateStopCluster`  methods

Reply:
🍾